torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -545,6 +551,16 @@ def symints_saved_for_backwards_slice(self):
         else:
             return slice(0, 0)  # empty slice
 
+    @property
+    def bw_donated_indices(self):
+        if self.fw_donated_buffer is None:
+            return []
+
+        # assuming backward input layout:
+        # [*symints, *saved_tensors, *flat_bw_args_with_grads, *rng_args].
+        assert self.num_symints_saved_for_bw is not None
+        return [self.num_symints_saved_for_bw + i for i in self.fw_donated_buffer]
```

Comment:
this code is all hot-path (it runs right before the compiled graph executes at runtime in the backward).

We've actually found that lots of list comprehensions noticeably impact the runtime overhead of AOTDispatcher. In general, the approach we've used is to pre-compute any indices that we need (here, that would mean pre-computing the exact indices of the backward inputs that we care about)